### DISCUSSION TITLE
65 vulnerabilities (1 low, 57 moderate, 7 high)

### DISCUSSION THREAD

**Discussion by edegraaff** [Koenkk/zigbee2mqtt#18106] [General]
Hi, i installed today Zigbee2MQTT and i was asking myself
Even after the update i get some risks in the log, how serious should i take this ?
48 moderate severity vulnerabilities
To address issues that do not require attention, run:
  npm audit fix
To address all issues possible (including breaking changes), run:
  npm audit fix --force
Some issues need review, and may require choosing
a different dependency.
Run `npm audit` for details.
npm notice
npm notice New major version of npm available! 8.19.4 -> 9.7.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.7.2
npm notice Run npm install -g npm@9.7.2 to update!
  @babel/core  *
  Depends on vulnerable versions of @babel/helper-compilation-targets
  Depends on vulnerable versions of semver
  node_modules/@babel/core
    @babel/helper-compilation-targets  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of semver
    node_modules/@babel/helper-compilation-targets
      @babel/plugin-transform-classes  >=7.19.0
      Depends on vulnerable versions of @babel/helper-compilation-targets
      node_modules/@babel/plugin-transform-classes
        @babel/preset-env

**Comment by lux73** [comment]
did you the npm update which is mentioned in the Logs and checked if your Problems are gone?

**Reply by edegraaff** [reply]
Hi, i have done the recommended update i am running npm notice New major version of npm available! 8.19.4 -> 9.7.2
root@zigbeemqtt:/opt/zigbee2mqtt# npm --version
9.7.2
root@zigbeemqtt:/opt/zigbee2mqtt#

git pull
> up to date

npm ci
65 vulnerabilities (1 low, 57 moderate, 7 high)

I also saw some messages regarding depricated modules
![image](https://github.com/Koenkk/zigbee2mqtt/assets/12442749/3d38b87c-5f8d-4229-8254-cea921ee9741)

I am just new to the project and try to get the whole running. 

If i do an 
npm audit fix --force
The result of  this action is in the log
My main question: should i take this kind of actions since it could break Zigbee2MQTT imho?

[report.txt](https://github.com/Koenkk/zigbee2mqtt/files/11872193/report.txt)

**Comment by lux73** [comment]
https://www.voitanos.io/blog/don-t-be-alarmed-by-vulnerabilities-after-running-npm-install/

**i would not do a** `npm audit fix --force` - it could break your z2m installation

in my opinion there is no problem with the "vulnerabilities" Warning

**Reply by edegraaff** [reply]
Thanks good to know, i will reinstall the PI. I am new to zigbee2mqtt so need to learn some do's and donts.

**Comment by YellowNest** [comment]
Since installing node 18.16.1 and npm v9.7.2 i get these errors

Installing dependencies...
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

added 1385 packages, and audited 1386 packages in 59s

81 packages are looking for funding
  run `npm fund` for details

83 vulnerabilities (1 low, 74 moderate, 8 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
Restore configuration...
Starting Zigbee2MQTT...
Done!

Everything is working fine though I see CPU is spiking frequently

**Reply by lux73** [reply]
try `npm rebuild` inside z2m Folder

https://www.w3resource.com/npm/npm-rebuild-repo-restart-root-and-run-script-commands.php

**Reply by YellowNest** [reply]
Npm rebuild didn’t work. Still left with 83 vulnerabilities

**Reply by lux73** [reply]
the "vulnerabilities" are no problem at all ^^

**Reply by YellowNest** [reply]
Since everything is working fine I assume that those warnings are no problem. The cpu usage is the only annoyance.

**Comment by edegraaff** [comment]
At my site everything is working fine on a pi3! thanks
